new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -132,16 +169,60 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
         mainWorkerPostReq,
     ]);
 
+    useEffect(() => {
+        logEventNumRef.current = logEventNum;
+    }, [logEventNum]);
+
+    useEffect(() => {
+        const newPage = (null === logEventNum || 0 >= logEventNum) ?
+            1 :
+            Math.max(1, numPages);
```

Comment:
Shouldn't this be something like:

```suggestion
        const newPage = clamp(null === logEventNum ?
            1 :
            Math.floor(logEventNum / PAGE_SIZE) + 1, 1, numPages);
```

Reply:
How about
```suggestion
        const newPage = null === logEventNum ?
            1 :
            clamp(Math.ceil(logEventNum / PAGE_SIZE), 1, numPages);
```

Reply:
Maybe we should extract a helper for calculations like `Math.ceil(logEventNum / PAGE_SIZE)` and also replace the `Math.floor((this.#numEvents - 1) / this.#pageSize)` expression in `LogFileManager.ts`.

Reply:
Sg